core/app/components/DataTable/PubsDataTable/PubsDataTableClient.tsx

```diff
@@ -0,0 +1,89 @@
+"use client";
+
+import React, { useMemo, useState } from "react";
+import { useQueryStates } from "nuqs";
+
+import type { ProcessedPub } from "contracts";
+import type { DataTableRowAction } from "ui/data-table-paged";
+import { DataTable, useDataTable } from "ui/data-table-paged";
+
+import { client } from "~/lib/api";
+import { useCommunity } from "../../providers/CommunityProvider";
+import { getColumns } from "./columns";
+import { dataTableParsers, getFilterParamsFromSearch } from "./validations";
+
+interface WithPromises {
+	promises: Promise<[number, ProcessedPub<{ withPubType: true }>[]]>;
+}
+interface WithData {
+	total: number;
+	data: ProcessedPub<{ withPubType: true }>[];
+}
+type PubsDataTableProps = {
+	perPage: number;
+} & (WithPromises | WithData);
+
+export const PubsDataTable = ({ perPage, ...props }: PubsDataTableProps) => {
+	let total = 0;
+	let data = [];
+	if ("promises" in props) {
+		const [t, d] = React.use(props.promises);
+		total = t;
+		data = d;
+	} else {
+		total = props.total;
+		data = props.data;
+	}
+	const pageCount = Math.ceil(total / perPage);
+	const [rowAction, setRowAction] = useState<DataTableRowAction<
+		ProcessedPub<{ withPubType: true }>
+	> | null>(null);
+	const columns = useMemo(() => getColumns({ setRowAction }), []);
+
+	const { table } = useDataTable({
+		data,
+		columns,
+		pageCount,
+		filterFields: undefined,
+		enableAdvancedFilter: false,
+		initialState: {
+			sorting: [{ id: "updatedAt", desc: true }],
+		},
+		getRowId: (originalRow) => originalRow.id,
+		shallow: false,
+		clearOnDefault: true,
+	});
+
+	return <DataTable table={table} floatingBar={null} />;
+};
+
+export const PubsDataTableClient = () => {
+	const [search, setSearch] = useQueryStates(dataTableParsers);
+	const community = useCommunity();
+	const filterParams = getFilterParamsFromSearch(search);
+
+	const { data } = client.pubs.getMany.useQuery({
+		queryKey: ["getPubs", filterParams],
+		queryData: {
+			query: {
+				...filterParams,
+				withPubType: true,
+				withRelatedPubs: false,
+				withStage: true,
+				withValues: false,
+				withLegacyAssignee: true,
+			},
+			params: {
+				communitySlug: community.slug,
+			},
+		},
+	});
```

Comment:
this currently is not being paginated right? or is that included in `filterParams`?

Reply:
yes, included in `filterParams`! I just wrapped it in `getFilterParamsFromSearch` so that both the server and the client side versions could get the same parameters